poppy/wfe.py

```diff
@@ -353,6 +353,231 @@ def get_opd(self, wave):
+                The PSD index value.
+            beta: astropy quantity
+                The normalization constant. In units of :math: `\frac{m^{2}}{m^{\alpha-2}}`
+                Numerator assumes surface units of meters
+                Denominator assumes spatial frequency units are 1/m
+            outer_scale: astropy quantity
+                The outer scale value, where the low spatial frequency flattens. 
+                Unit requirement: meters
+            inner_scale: float
+                Inner scale value, where the high spatial frequency flattens.
+            surf_roughness: astropy quantity
+                Surface roughness normalization. Should match units of PSD.
+    psd_weight: iterable list of floats
+        Specifies the weight muliplier to set onto each model PSD
+    seed : integer
+        Seed for the random phase screen generator
+    apply_reflection: boolean
+        Applies 2x scale for the OPD as needed for reflection.
+        Default to False. 
+        Set to True if the PSD model only accounts for surface.
+    screen_size: integer
+        Sets how large the PSD matrix will be calculated.
+        The PSD matrix needs to be larger than the wavefront for Fourier transform padding purposes.
+        If None passed in, then code will default size to 4x wavefront's side.
+        Default to None.
+    rms: astropy quantity
+        Optional. Use this to force the wfe RMS
+        If a value is passed in, this is the surface rms value (not OPD) in meters.
+        If None passed, then the wfe RMS produced is what shows up in PSD calculation.
+        Default to None.
+    incident_angle: astropy quantity
+        Adjusts the WFE based on 
+        Can be passed as either degrees or radians.
+        Default is 0 degrees (paraxial).
+    """
+
+    @utils.quantity_input(wfe=u.nm, radius=u.meter)
+    def __init__(self, name='Model PSD WFE', psd_parameters=None, psd_weight=None, seed=None, 
+                 apply_reflection=False, screen_size=None, rms=None, incident_angle=0*u.deg, 
+                 **kwargs):
+
+        super().__init__(name=name, **kwargs)
+        self.psd_parameters = psd_parameters
+        self.seed = seed
+        self.apply_reflection = apply_reflection
+        self.screen_size = screen_size
+        self.rms = rms
+        
+        # check incident angle units
+        if hasattr(incident_angle, 'unit'):
+            if incident_angle.unit == u.deg:
+                assert incident_angle.value < 90, "Incident angle must be less than 90 degrees."
+            elif incident_angle.unit == u.rad:
+                assert incident_angle.value < (np.pi/2), "Incident angle must be less than pi/2 radians."
+            else:
+                raise Exception('Incident angle units must be either degrees or radians.')
+            self.incident_angle = incident_angle
+        else:
+            raise Exception('Incident angle missing units, must be either degrees or radians.')
```

Comment:
This piece can be automated by adding it to the quantity input wrapper above the function. Add incident angle to that, like `@utils.quantity_input(wfe=u.nm, radius=u.meter, incident_angle=u.deg)`. It will automatically enforce that any input to incident angle has to be in degrees or some other equivalent convertible unit (radians, arcmin, anything angular). 

That won't apply the range check relative to 90º, but that can also be simplified using units too, like this: 
```
if incident_angle >= 90*u.deg: 
    raise ValueError("Incident angle must be less than 90 degrees, or equivalent in other units")
```
That relies on the units framework to handle the conversions. If incident angle is given in radians or any other angular unit, when it's compared to a value in degrees the conversion will happen automatically.

Reply:
Wow that's cool! Once again you're finding more of my rudimentary python coding. I made the fix, thanks for the tip.